examples/conform.cpp

```diff
@@ -0,0 +1,164 @@
+//
+// Copyright Contributors to the OpenTimelineIO project
+//
+// Licensed under the Apache License, Version 2.0 (the "Apache License")
+// with the following modification; you may not use this file except in
+// compliance with the Apache License and the following modification to it:
+// Section 6. Trademarks. is deleted and replaced with:
+//
+// 6. Trademarks. This License does not grant permission to use the trade
+//    names, trademarks, service marks, or product names of the Licensor
+//    and its affiliates, except as required to comply with Section 4(c) of
+//    the License and to reproduce the content of the NOTICE file.
+//
+// You may obtain a copy of the Apache License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the Apache License with the above modification is
+// distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the Apache License for the specific
+// language governing permissions and limitations under the Apache License.
+//
+
+// Example OTIO script that reads a timeline and then relinks clips
+// to movie files found in a given folder, based on matching names.
+//
+// Demo:
+//
+// % ls -1R
+// editorial_cut.otio
+// media/
+//    shot1.mov
+//    shot17.mov
+//    shot99.mov
+//
+// % conform editorial_cut.otio media conformed.otio
+// Relinked 3 clips to new media.
+// Saved conformed.otio with 100 clips.
+//
+// % diff editorial_cut.otio conformed.otio
+// ...
+
+#include "util.h"
+
+#include <opentimelineio/clip.h>
+#include <opentimelineio/externalReference.h>
+#include <opentimelineio/timeline.h>
+
+#include <iostream>
+
+namespace otio = opentimelineio::OPENTIMELINEIO_VERSION;
+
+// Look for media with this name in this folder.
+std::string find_matching_media(std::string const& name, std::string const& folder)
+{
+    // In this case we're looking in the filesystem.
+    // In your case, you might want to look in your asset management system
+    // and you might want to use studio-specific metadata in the clip instead
+    // of just the clip name.
+    // Something like this:
+    // shot = asset_database->find_shot(
+    //    otio::any_cast<std::map<std::string, std::string> >(clip->metadata()["mystudio"])["shotID"]);
+    // new_media = shot->latest_render("mov");
```

Comment:
Light editing

```suggestion
    // This function is an example which searches the file system for matching media.
    // A real world studio implementation would likely look in an asset management system
    // and use studio-specific metadata in the clip's metadata dictionary instead
    // of matching the clip name.
    // For example:
    // shot = asset_database->find_shot(
    //    otio::any_cast<std::map<std::string, std::string> >(clip->metadata()["mystudio"])["shotID"]);
    // new_media = shot->latest_render("mov");
```

Reply:
Looks good, this text was taken from the Python example so I'll make the changes there as well.